src/main/java/seedu/address/logic/commands/room/SearchRoomCommand.java

```diff
@@ -32,20 +35,17 @@ public SearchRoomCommand(Integer roomNumber) {
         this.roomNumber = roomNumber;
     }
 
-    //TODO
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        /*Index index = model.getModifiableRoomList().checkIfRoomPresent(roomNumber);
-        List<Room> lastShownList = model.getModifiableRoomList().getRoomObservableList();
+        requireNonNull(model);
 
+        Index index = model.checkIfRoomPresent(roomNumber);
         if (index.getZeroBased() == 0) {
             throw new CommandException(MESSAGE_INVALID_ROOM_NOT_FOUND);
         }
 
-        Room roomToEdit = lastShownList.get(index.getZeroBased() - 1);
-
-         */
-        return null;
+        model.updateFilteredRoomList(room -> room.getRoomNumber() == roomNumber);
```

Comment:
Maybe can include the predicate in a different method like:
private Predicate<Room> getPredicateForSearchRoom(int roomNumberExpected) {
           Predicate<Room> predicateForSearchRoom = room1 -> room1.getRoomNumber() == roomNumberExpected;
           return predicateForSearchRoom;
}
That is wat I meant.